### PR TITLE
Added support in Version upgrade test framework to test upgrade with non-default-server-collation

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
This commit adds a support in version upgrade test framework to perform a upgrade with non-default server collation.
This also adds new workflow to run normal jdbc test framework with non-default server collation which will run all the
jdbc tests with non-default server collation name.

### Upgrade paths added to test upgrade with non-default server collation:
1. [13.6 --> 14.latest --> 15.latest --> 16.latest] with server collation name: chinese_prc_ci_as
2. [14.6 --> 16.latest] with server collation name: chinese_prc_ci_as
3. [15.6 --> 16.latest] with server collation name: chinese_prc_ci_as
4. [15.6 --> 16.latest] with server collation name: japanese_ci_as
5. [15.5 --> 15.6 --> 16.latest] with server collation name: chinese_prc_ci_as
6. [16.1 --> 16.latest] with server collation name: chinese_prc_ci_as

### JDBC Test with non-default server collation:
In this workflow, JDBC test will run with non-default server collation such as chinese_prc_ci_as. (Note: server collation can be changed by modifying SERVER_COLLATION_NAME environment variable to desired server collation in workflow file for this test)

Cherry-picked from PR: [PR-2096](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2096), [PR-2311](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2311)
Signed-off-by: Rohit Bhagat <rohitbgt@amazon.com>

### Issues Resolved

BABEL-4590

### Test Scenarios Covered ###
* **Use case based -** NA

* **Boundary conditions -** NA

* **Arbitrary inputs -** NA

* **Negative test cases -** NA

* **Minor version upgrade tests -** NA

* **Major version upgrade tests -** NA

* **Performance tests -** NA

* **Tooling impact -** NA

* **Client tests -** NA

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/composite-actions/major-version-upgrade-util/action.yml [hunk 1]
```diff
@@ -39,11 +43,13 @@ runs:
...
     - name: Run Verify Tests
       if: always() && steps.run-pg_upgrade.outcome == 'success' && inputs.is_final_ver == 'true'
       uses: ./.github/composite-actions/run-verify-tests
       with:
         is_final_ver: ${{ inputs.is_final_ver }}
         pg_new_dir: ${{ inputs.pg_new_dir }}
-        migration_mode: ${{ inputs.migration_mode }}
\ No newline at end of file
+        migration_mode: ${{ inputs.migration_mode }}
+        server_collation_name: ${{ inputs.server_collation_name }}
```
Comment (rishabhtanwar29) [inline .github/composite-actions/major-version-upgrade-util/action.yml]: Nit: let's have a newline at the end of the file.

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql [hunk 2]
```diff
@@ -11,6 +11,26 @@ SELECT set_config('search_path', 'sys, '||current_setting('search_path'), false)
  * final behaviour.
  */
 
+CREATE OR REPLACE FUNCTION sys.babelfish_update_server_collation_name() RETURNS VOID
+LANGUAGE C
+AS 'babelfishpg_common', 'babelfish_update_server_collation_name';
+
+SELECT sys.babelfish_update_server_collation_name();
+
+DROP FUNCTION sys.babelfish_update_server_collation_name();
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql:20]: Do we really need this ? Original comment says that it is only needed in 2.3->3.x https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_4_X_DEV/contrib/babelfishpg_common/src/collation.c#L1585-L1586

File: test/JDBC/expected/BABEL-UNSUPPORTED.out [hunk 3]
```diff
@@ -527,6 +527,12 @@ GO
 
 ~~ERROR (Message: only "sql_latin1_general_cp1_ci_as" supported for default collation)~~
 
+DROP DATABASE db_unsupported9;
+GO
+~~ERROR (Code: 911)~~
+
+~~ERROR (Message: database "db_unsupported9" does not exist)~~
```
Comment (rishabhtanwar29) [inline test/JDBC/expected/BABEL-UNSUPPORTED.out:534]: Why do we need this?

File: .github/workflows/jdbc-tests-with-non-default-server-collation.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      - name: Upload Output Diff
+        if: always() && steps.jdbc.outcome == 'failure'
+        uses: actions/upload-artifact@v2
+        with:
+          name: output-diff.diff
+          path: test/JDBC/Info/output-diff.diff
+
+      - name: Check and upload coredumps
+        if: always() && steps.jdbc.outcome == 'failure'
+        uses: ./.github/composite-actions/upload-coredump
```
Comment (rishabhtanwar29) [inline .github/workflows/jdbc-tests-with-non-default-server-collation.yml]: Nit: add newline

File: .github/configuration/upgrade-test-configuration.yml [hunk 5]
```diff
@@ -1,116 +1,210 @@
...
+    },
+    { 
+      version: target.latest,
+      upgrade-type: major
+    }
+  ],
+  babelfishpg_tsql_server_collation_name: chinese_prc_ci_as
+},
+{
+  upgrade-path: [
```
Comment (rishabhtanwar29) [inline .github/configuration/upgrade-test-configuration.yml:185]: Let's have one more path from 15.5 -> 15.6 -> latest

File: .github/workflows/jdbc-tests-with-non-default-server-collation.yml [hunk 6]
```diff
@@ -0,0 +1,111 @@
+name: JDBC Tests with Non Default Server Collation
```
Comment (rishabhtanwar29) [inline .github/workflows/jdbc-tests-with-non-default-server-collation.yml:1]: I have added logical replication support with JDBC framework, let's include those changes in this as well. Refer https://github.com/babelfish-for-postgresql/babelfish_extensions/commit/214ea5831d3077bb900ac1753609364a3edae4c9

File: .github/composite-actions/major-version-upgrade-util/action.yml (hunk 1 above)
Comment (rohit01010) [reply .github/composite-actions/major-version-upgrade-util/action.yml]: done

File: .github/configuration/upgrade-test-configuration.yml (hunk 5 above)
Comment (rohit01010) [reply .github/configuration/upgrade-test-configuration.yml:185]: Added

File: .github/workflows/jdbc-tests-with-non-default-server-collation.yml (hunk 6 above)
Comment (rohit01010) [reply .github/workflows/jdbc-tests-with-non-default-server-collation.yml:1]: Done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql (hunk 2 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql:20]: the comment says `This helper function would only be useful and strictly be used during 1.x->2.3 and 2.3->3.0 upgrade`.
I believe this comment was intended to say that this function execution should be strictly be used in all major version upgrade scripts, and as back then version 4.0 was not available its written only till 3.x.
cc: @Deepesh125

File: test/JDBC/expected/BABEL-UNSUPPORTED.out (hunk 3 above)
Comment (rohit01010) [reply test/JDBC/expected/BABEL-UNSUPPORTED.out:534]: In this test, db_unsupported9 is being created and is not dropped, this is causing interference with other test files when server collation name is `Arabic_CI_AS`

File: .github/composite-actions/major-version-upgrade-util/action.yml (hunk 1 above)
Comment (rohit01010) [reply .github/composite-actions/major-version-upgrade-util/action.yml]: done

File: .github/workflows/jdbc-tests-with-non-default-server-collation.yml (hunk 4 above)
Comment (rohit01010) [reply .github/workflows/jdbc-tests-with-non-default-server-collation.yml]: done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql (hunk 2 above)
Comment (Deepesh125) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.5.0--4.0.0.sql:20]: yes, it should be applicable to 2.3 to all major version upgrade

**Review by Deepesh125 — DISMISSED** [review]

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**